### PR TITLE
kctrl: Remove `install` option in package installed update command

### LINKED ISSUE #678: `kctrl`: Remove the `--install` and `--create` flags for package installed update and package repository update
**Describe the problem/challenge you have**
`package repository add` currently allows to both add or update a package repository and the `create` option in update seems redundant.
Similarly, the `install` option in package installed update is redundant.

**Describe the solution you'd like**
[A clear and concise description of what you want to happen. If applicable a visual representation of the UX.]
- Remove the `create` option in `package repository update`, i.e. the command should only allow updates and not creation.
- Remove the `install` option in `package installed update`, i.e. the command should only allow updating an installation and not creation of a new installation.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### LINKED ISSUE #679: `kctrl`: `package installed update` should have proper logging or error message when there is nothing to update.
**What steps did you take:**
`kctrl package installed create -i test-pkg -p cert-manager --version 1.0.0`
`kctrl package installed update -i test-pkg`

**What happened:**
[A small description of the issue]
`Getting package install for 'test-pkg'`

**What did you expect:**
[A description of what was expected]
An error message saying that there's nothing to update, or logs updating the package.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

**Environment:**

- kapp Controller version (execute `kubectl get deployment -n kapp-controller kapp-controller -o yaml` and the annotation is `kbld.k14s.io/images`):
- Kubernetes version (use `kubectl version`)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### LINKED ISSUE #682: `kctrl` package installed update creates a package install with empty package name
**What steps did you take:**
```bash
$ kctrl package installed update -i test-carvel-pkg --version 3.0.0-rc.1 -n test-ns --install
Getting package install for 'test-carvel-pkg'
Updating package install for 'test-carvel-pkg'
Waiting for PackageInstall reconciliation for 'test-carvel-pkg'
3:46:46PM: packageinstall/test-carvel-pkg (packaging.carvel.dev/v1alpha1) namespace: test-ns: ReconcileFailed
Error: packageinstall/test-carvel-pkg (packaging.carvel.dev/v1alpha1) namespace: test-ns: Reconciling: . Reconcile failed: Expected to find at least one version, but did not (details: a...
```

```bash
$ kctrl package installed update -i test-carvel-pkg -p pkg.test.carvel.dev  --version 3.0.0-rc.1 -n test-ns --install
Getting package install for 'test-carvel-pkg'
Error: Installed package 'test-carvel-pkg' is already associated with package ''
```

**What happened:**
kctrl created a package install with empty package name when using `--install` flag with `package installed update` command and not providing a package name.

**What did you expect:**
Error saying package name is required.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]
This should be resolved with #678, but creating an issue to make sure that it gets tracked.

**Environment:**

- kapp Controller version (execute `kubectl get deployment -n kapp-controller kapp-controller -o yaml` and the annotation is `kbld.k14s.io/images`):
- Kubernetes version (use `kubectl version`)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by praveenrewar**
#### What this PR does / why we need it:
- Remove `install` option in package installed update command as package installed create can be used to install or update a package)
- Remove `create` option for package repository update as package repository add can be used to create or update a repository.
- Update error messages to be more helpful.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
`package installed update` doesn't support `--install` option. `package installed create` can still be used for installing as well as updating existing package installs.
`package repository update` doesn't support `--create` option. `package repository create` can still be used for adding or updating existing package repositories.
```

##### Review Checklist:

- [x] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [x] Relevant tests are added or updated
- [x] Relevant docs in this repo added or updated
- [x] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [x] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: cli/test/e2e/package_install_test.go [hunk 1]
```diff
@@ -191,6 +192,9 @@ spec:
 		}, RunOpts{})
 		require.NoError(t, err)
 
+		// Sleep for 1 second as it takes time for the underlying app cr to be paused
+		time.Sleep(1 * time.Second)
```
Comment (100mik) [inline cli/test/e2e/package_install_test.go:196]: Wondering if we should add waiting logic for pause, but I do not think it makes sense
We could go with this for now

**Review by 100mik — APPROVED** [review]
LGTM 🚀 
(just added one thought)

File: cli/pkg/kctrl/cmd/package/installed/create_or_update.go [hunk 2]
```diff
@@ -159,10 +159,8 @@ func NewUpdateCmd(o *CreateOrUpdateOptions, flagsFactory cmdcore.FlagsFactory) *
 		cmd.Args = cobra.ExactArgs(1)
 	}
 
-	cmd.Flags().StringVarP(&o.packageName, "package", "p", "", "Name of package install to be updated")
```
Comment (cppforlife) [inline cli/pkg/kctrl/cmd/package/installed/create_or_update.go:162]: how come package is removed? it's valid to change PackageInstall to use a different package name. (though probably less common). may be it should be --dangerous-package?

Comment (praveenrewar) [reply cli/pkg/kctrl/cmd/package/installed/create_or_update.go:162]: Hmm, makes sense. I will revert this commit for now. 
cc @100mik

File: cli/pkg/kctrl/cmd/package/repository/add_or_update.go [hunk 3]
```diff
@@ -74,7 +74,6 @@ func NewAddCmd(o *AddOrUpdateOptions, flagsFactory cmdcore.FlagsFactory) *cobra.
 		DefaultTimeout:   5 * time.Minute,
 	})
 
-	// For `add` command create option will always be true
 	o.CreateRepository = true
```
Comment (100mik) [inline cli/pkg/kctrl/cmd/package/repository/add_or_update.go:77]: So are we using this variable to determine whether or not the run command should update or add? Seems like we could do this better, but definitely not blocking.

We should however change the check on line 138 to use the `errors` library in this PR imo.
Or do yopu wanna do this in another PR?

**Review by 100mik — COMMENTED** [review]
Just a few thoughts

File: cli/pkg/kctrl/cmd/package/repository/add_or_update.go (hunk 3 above)
Comment (praveenrewar) [reply cli/pkg/kctrl/cmd/package/repository/add_or_update.go:77]: I wanted to keep the changes in this PR to a minimum. We can definitely do some refactoring in later PRs which do not change the behaviour for users.

Comment (100mik) [reply cli/pkg/kctrl/cmd/package/repository/add_or_update.go:77]: These will be going in with the next PR as discussed on slack
